File: extras/server/prompts/prompt-builder.js
// server/prompts/prompt-builder.js
import { SYSTEM_PROMPTS } from './system-prompts.js';

export function buildSQLGenerationPrompt(userQuery, schemaContext, knowledge, clarifications) {
  let prompt = SYSTEM_PROMPTS.SQL_GENERATION;

  prompt += '\n\n## DATABASE SCHEMA\n\n';
  prompt += schemaContext;

  if (knowledge && knowledge.length > 0) {
    prompt += '\n\n## DOMAIN KNOWLEDGE\n\n';
    prompt += knowledge;
  }

  if (clarifications && Object.keys(clarifications).length > 0) {
    prompt += '\n\n## USER CLARIFICATIONS\n\n';
    for (const [question, answer] of Object.entries(clarifications)) {
      prompt += `Q: ${question}\nA: ${answer}\n\n`;
    }
  }

  prompt += '\n\n## USER QUERY\n\n';
  prompt += userQuery;

  return prompt;
}

// Ähnlich: buildAmbiguityDetectionPrompt, buildKnowledgeSearchPrompt,
// buildValidationPrompt, buildExplanationPrompt

File: backend/app/prompts/prompt_builder.py
from typing import Dict


def build_sql_generation_prompt(user_query: str, schema_context: str, knowledge: str, clarifications: Dict) -> str:
    """Compose the prompt text for SQL generation."""
    clarifications_text = ""
    if clarifications:
        clarifications_text = f"Clarifications: {clarifications}"

    return f"""You are to generate a SQLite SELECT query.

User Query:
{user_query}

Database Schema:
{schema_context}

Relevant Knowledge:
{knowledge}

{clarifications_text}
"""
